### PR TITLE
doc: matter: add standalone security page

### PR DESCRIPTION
**Pull request by greg-fer**
Separated the security section from the network overview page. Added standalone security page.
KRKNWK-15223.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]

File: doc/nrf/ug_matter_overview_architecture.rst [hunk 1]
```diff
@@ -88,5 +88,6 @@ Transport and IP Framing
 ************************
 
 This layer manages the transmission of the payload through the IP network to the peer device.
-It uses either the Transmission Control Protocol (TCP) or the combination of User Datagram Protocol (UDP) with Matter's Message Reliable Protocol (MRP).
-During the commissioning process, Bluetooth Transport Protocol over Bluetooth LE can be used instead of that layer.
+It uses either the Transmission Control Protocol (TCP) or the combination of User Datagram Protocol (UDP) with Matter's Message Reliability Protocol (MRP).
+The MRP ensures that duplicated messages are rejected.
```
Comment (markaj-nordic) [inline doc/nrf/ug_matter_overview_architecture.rst:92]: I guess it's not only that, right (i.e. MRP implements e.g. retransmissions)?

File: doc/nrf/ug_matter_device_certification.rst [hunk 2]
```diff
@@ -41,6 +41,22 @@ Matter certification requires testing performed at Authorized Test Laboratory (A
...
+For a Matter end product to be certified, CSA's policies require an Attestation of Security that provides detailed information about the security level of the end product.
+The attestation document lists robustness security requirements based on the Matter specification.
+The product developer must indicate the level of compliance and briefly justify the choice.
+
+The attestation must be filled by the person responsible for end product certification who meets the following requirements:
+
+* The person's organization is a `member of the Matter community <Join CSA_>`_.
+* The person has an account on the `Connectivity Standards Alliance Certification Web Tool`_.
+
+Once both these requirements are met, the responsible person can download the `Matter Attestation of Security template`_, fill it in, and submit it in the certification web tool when applying for the certification.
```
Comment (markaj-nordic) [inline doc/nrf/ug_matter_device_certification.rst:58]:
```suggestion
Once both these requirements are met, the responsible person can download the `Matter Attestation of Security template`_, fill it in, and submit it in the certification web tool when applying for the attestation.
```
Not sure about that though (looks like the section refers to the attestation which is a part of certification process).

**Review by markaj-nordic — APPROVED** [review]

File: doc/nrf/ug_matter_overview_architecture.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_overview_architecture.rst:92]: Rephrased to "The MRP implements retransmissions, provides confirmation of delivery, and ensures that duplicated messages are rejected."

File: doc/nrf/ug_matter_device_certification.rst (hunk 2 above)
Comment (doublemis1) [reply doc/nrf/ug_matter_device_certification.rst:58]: The certification is correct.
The user will apply for certification,
The attestation of some End Product aspects is a part of Certification documentation.

Comment (doublemis1) [inline doc/nrf/ug_matter_device_certification.rst:58]: In all documents on CSA there is a Matter Security Attestation template. I will stick with that name to have the same naming as CSA.
`Matter Attestation of Security template`

File: doc/nrf/ug_matter_device_certification.rst [hunk 3]
```diff
@@ -41,6 +41,22 @@ Matter certification requires testing performed at Authorized Test Laboratory (A
 
 To read more about Matter and dependent certification, get familiar with CSA's Certification Policy and Procedures document.
 
+.. _ug_matter_device_certification_reqs_security:
+
+Matter Attestation of Security
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/ug_matter_device_certification.rst:46]: The CSA's document template uses "Matter Attestation of Security".

File: doc/nrf/ug_matter_device_certification.rst (hunk 2 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_device_certification.rst:58]: The CSA's document template uses "Matter Attestation of Security".

**Review by doublemis1 — APPROVED** [review]
LGTM!

**Review by kkasperczyk-no — APPROVED** [review]

**Pull request merged**